src/MoBi.Core/Domain/Services/SimulationFactory.cs

```diff
@@ -105,10 +108,15 @@ public IModel CreateModelAndValidate(SimulationConfiguration simulationConfigura
          return results.Model;
       }
 
-      public IMoBiSimulation CreateSimulationAndValidate(SimulationConfiguration configuration, string simulationName)
+      public IMoBiSimulation CreateSimulationAndValidate(SimulationConfiguration configurationReferencingBuildingBlocks, string simulationName)
       {
-         var model = CreateModelAndValidate(configuration, simulationName, AppConstants.Captions.CreatingSimulation);
-         return createSimulation(model, configuration, simulationName);
+         var model = CreateModelAndValidate(configurationReferencingBuildingBlocks, simulationName, AppConstants.Captions.CreatingSimulation);
+         return CreateFrom(cloneOf(configurationReferencingBuildingBlocks), model).WithName(simulationName);
+      }
+
+      private SimulationConfiguration cloneOf(SimulationConfiguration configurationReferencingBuildingBlocks)
+      {
+         return _cloneManager.Clone(configurationReferencingBuildingBlocks);
```

Comment:
simlarily here. I don't think having a method that calls the method of one member makes it more readable, especially because used once. Personal preference, but the cloneManage.clone is as readable of cloneOf. Adding a method is just a level of indirection